### PR TITLE
app-metrics/prometheus-podman-exporter: add 1.4.0 & live

### PR DESCRIPTION
**Pull request by rahilarious**
I'll add openrc's init.d script if someone provides patch/PR

### CODE REVIEW

**Comment by rahilarious** [comment]
@zmedico any input on this one?

**Review by zmedico — APPROVED** [review]
Looks good! Thanks!

**Comment by rahilarious** [comment]
> Looks good! Thanks!

May you merge please?

**Comment by zmedico** [comment]
> > Looks good! Thanks!
> 
> May you merge please?

I'm not a member of @gentoo/proxy-maint so let's let them merge or at least give me an ack.

File: app-metrics/prometheus-podman-exporter/prometheus-podman-exporter-1.4.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+else
+	SRC_URI="https://github.com/containers/prometheus-podman-exporter/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+# Main package
+LICENSE="Apache-2.0"
+# Dependencies
+LICENSE+=" BSD-2 BSD ISC MIT MPL-2.0"
+SLOT="0"
+RESTRICT="test"
```
Comment (juippis) [inline app-metrics/prometheus-podman-exporter/prometheus-podman-exporter-1.4.0.ebuild:22]: Please add a comment stating why tests are restricted. Link to upstream issue if there's one open.

File: app-metrics/prometheus-podman-exporter/prometheus-podman-exporter-1.4.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+RDEPEND="
+	>=app-containers/podman-4.0.0
+"
+DEPEND="${RDEPEND}"
```
Comment (juippis) [inline app-metrics/prometheus-podman-exporter/prometheus-podman-exporter-1.4.0.ebuild:26]: Nitpick: add some whitespaces, aka follow the ebuild.skel structure a bit more closely here :P

File: app-metrics/prometheus-podman-exporter/prometheus-podman-exporter-1.4.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+src_compile() {
+	export BUILDFLAGS=" -tags exclude_graphdriver_devicemapper"
+	has_version app-containers/podman[-btrfs] && BUILDFLAGS+=",exclude_graphdriver_btrfs,btrfs_noversion"
```
Comment (juippis) [inline app-metrics/prometheus-podman-exporter/prometheus-podman-exporter-1.4.0.ebuild:30]: I like this style, but this isn't truly a working solution. If you flip the `btrfs` flag on podman, nothing will happen for this package. Here you want to introduce a use flag and control it with `app-containers/podman[btrfs=]`.

**Review by juippis — COMMENTED** [review]
Comments apply to -9999 as well.

File: app-metrics/prometheus-podman-exporter/prometheus-podman-exporter-1.4.0.ebuild (hunk 1 above)
Comment (rahilarious) [reply app-metrics/prometheus-podman-exporter/prometheus-podman-exporter-1.4.0.ebuild]: done

File: app-metrics/prometheus-podman-exporter/prometheus-podman-exporter-1.4.0.ebuild (hunk 2 above)
Comment (rahilarious) [reply app-metrics/prometheus-podman-exporter/prometheus-podman-exporter-1.4.0.ebuild]: done

File: app-metrics/prometheus-podman-exporter/prometheus-podman-exporter-1.4.0.ebuild (hunk 3 above)
Comment (rahilarious) [reply app-metrics/prometheus-podman-exporter/prometheus-podman-exporter-1.4.0.ebuild]: Done. But implemented with different logic. If btrfs use is enabled then force podman[btrfs]  else just depend on podman

**Comment by dlan17** [comment]
any particular good reason that you want to push the live ebuild to official portage tree?
if answer is NO, then I strongly recommend dropping it as it will cost extra effort for proxy maintainers..

**Comment by rahilarious** [comment]
> any particular good reason that you want to push the live ebuild to official portage tree? if answer is NO, then I strongly recommend dropping it as it will cost extra effort for proxy maintainers..

Being the proxied maintainer of this package, I run live version on my system so that whenver something breaks I can immediately report upstream and try to get it fixed. Most upstreams require user to be on HEAD. Whenever new version is released I can easily bump by copying. 

I'll make sure there will be no extra effort for p-m, as I'm elaborative in commits. I already maintain quite few [packages](https://packages.gentoo.org/maintainer/rahil3108@gmail.com)  this way with live versions

**Comment by dlan17** [comment]
> > any particular good reason that you want to push the live ebuild to official portage tree? if answer is NO, then I strongly recommend dropping it as it will cost extra effort for proxy maintainers..
> 
> Being the proxied maintainer of this package, I run live version on my system so that whenver something breaks I can immediately report upstream and try to get it fixed. Most upstreams require user to be on HEAD. Whenever new version is released I can easily bump by copying.
> 
I'm not convinced by this reason, it's ONLY useful for maintainer, not for users.

> I'll make sure there will be no extra effort for p-m, as I'm elaborative in commits. I already 
no extra effort, can't 100% sure about this..

>maintain quite few [packages](https://packages.gentoo.org/maintainer/rahil3108@gmail.com) >this way with live versions
still not direct reason for pushing the live version, as a compromise, would be it ok for you to drop the live ebuild while still keeping the logic inside version 1.4.0, so whenever you need, you can copy & paste to live version and test it, or just host the live ebuild inside you own local overlay?

**Comment by rahilarious** [comment]
> > still not direct reason for pushing the live version, as a compromise, would be it ok for you to drop the live ebuild while still keeping the logic inside version 1.4.0, so whenever you need, you can copy & paste to live version and test it, or just host the live ebuild inside you own local overlay?

Sure I could do that if you insist but please let me know the dangers/drawbacks of having live ebuild in repo? Users will have to manually change ACCEPT_KEYWORDS to ** to enable.

**Comment by dlan17** [comment]
> > > still not direct reason for pushing the live version, as a compromise, would be it ok for you to drop the live ebuild while still keeping the logic inside version 1.4.0, so whenever you need, you can copy & paste to live version and test it, or just host the live ebuild inside you own local overlay?
> 
> Sure I could do that if you insist but please let me know the dangers/drawbacks of having live ebuild in repo? Users will have to manually change ACCEPT_KEYWORDS to ** to enable.

ok, the drawbacks:
1) the state of live ebuild usually is volatile, which mean the source changes constantly, and would become broken easily, this would require additional PR/fix? then it would require extra effort from Gentoo proxy maintainer side to push/review? I'd hope we could avoid this & reduce the effort given there isn't enough manpower inside the proxymaint team..
(bottom line: if you could find some else willing to proxy to take care of live ebuild, then I'd be fine)
2) we shouldn't really encourage user to use live ebuild?

**Comment by rahilarious** [comment]
>   1. the state of live ebuild usually is volatile, which mean the source changes constantly, and would become broken easily, this would require additional PR/fix? then it would require extra effort from Gentoo proxy maintainer side to push/review? I'd hope we could avoid this & reduce the effort given there isn't enough manpower inside the proxymaint team..
>      (bottom line: if you could find some else willing to proxy to take care of live ebuild, then I'd be fine)

Agreed. Though in case of this package, it's a Go package which would hardly need updating live. 

>   2. we shouldn't really encourage user to use live ebuild?

Why not? Gentoo is about choices. If user wants to on bleeding edge, then let them. Their own blood, their choice to bleed xD.

**Comment by rahilarious** [comment]
> Let's give it a try. We generally dislike -9999s because they're always left behind to rot, and because proxy-maint project is a bit slow to respond we can't be fixing problems in a -9999 immediately. And overall -9999s aren't very user-friendly. But if you use it and keep it maintained, why not. Just know we have a very low barrier in removing these -9999s if they cause problems.
> 
> Thanks!

Understood. I personally treat 9999s as first class citizens. Thank you for putting trust in me. Cheers

**Pull request closed without merging**